pkg/apis/compliance/v1alpha1/scansetting_types.go

```diff
@@ -0,0 +1,44 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ScanSettingSpec defines the desired state of ScanSetting
+type ScanSettingSpec struct {
+	// Defines whether or not the remediations should be applied automatically
+	AutoApplyRemediations bool `json:"autoApplyRemediations,omitempty"`
+	// Defines a schedule for the scans to run. This is in cronjob format.
+	// Note the scan will still be triggered immediately, and the scheduled
+	// scans will start running only after the initial results are ready.
+	Schedule string `json:"schedule,omitempty"`
+	// Disables cleaning up resources in the DONE phase, this might be useful for debugging.
+	Debug bool `json:"debug,omitempty"`
+	// The list of roles to apply node-specific checks to
+	Roles []string `json:"roles,omitempty"`
+}
```

Comment:
What about adding this to the ComplianceScan and replacing the values we have there? That way the settings will always be up-to-date

Reply:
Sorry, I haven't seen this comment before I pushed the new version. I'll think about this tomorrow. In the meantime, the current version is no longer WIP and has a test (although I guess more e2e tests should be added...)

Reply:
At this point, with all the new parameters that have been added to the scan, I think it's necessary to embed this structure there and update the parameters

Reply:
I guess you're right, code duplication is bad.

Reply:
This is done is a separate commit as the existing controllers also had to be changed.